src/plugins/mpi/pmix/mpi_pmix.c

```diff
@@ -159,6 +162,12 @@ extern int p_mpi_hook_slurmstepd_prefork(
 		PMIXP_ERROR("pmixp_stepd_init() failed");
 		goto err_ext;
 	}
+
+	if (SLURM_SUCCESS != (ret = pmixp_abort_agent_start(env))) {
```

Comment:
This should be executed from `p_mpi_hook_client_prelaunch()`.
The way it is now, abort agent resides inside stepd.

Reply:
if we called this - we are stepd